main.go

```diff
@@ -133,24 +135,37 @@ func main() {
 		os.Exit(1)
 	}
 
-	// TODO: Get a Tinkerbell client.
+	tinkerbellClient, err := tinkerbell.NewClient(ctrl.Log.WithName("TinkerbellClient"))
+	if err != nil {
+		setupLog.Error(err, "creating Tinkerbell client")
+
+		os.Exit(1)
+	}
 
 	if webhookPort == 0 {
+		crcc := &reconcilers.ClusterReconcileContextConfig{
+			Client:           mgr.GetClient(),
+			HardwareIPGetter: tinkerbellClient,
+			Log:              ctrl.Log.WithName("controllers").WithName("TinkerbellCluster"),
+		}
```

Comment:
My idea here was to have a distinction between the `TinkerbellMachineReconciler`, which is a controller and handles `Reconcile()` method. As handling both cluster and machine objects is rather complex (e.g. we need to pull plenty of dependent objects before we can start processing), I think it's a good idea to separate this controller from the object which handles just a single cluster/machine.

With separate object we can express "Given the following data sources/sinks (dependencies like Client, Tinkerbell Client etc.) reconcile me a cluster/machine with a given name (namespaced name)". This way, we can build up an API used by the controller, so it's responsibilities are minimized to what controller should do, for example it can keep the control over when to re-queue the reconciliation (like `New()` or `IntoMachineReconcileContext()` which may return `nil` when dependencies are not ready yet).

If we move config bits into `TinkerbellClusterReconciler`, then we must either extend the API of this struct or copy the fields into separate config in `Reconcile()` method if we want to keep the abstraction/separation I described.

Having those 2 separate should also allow easier testing, when we turn clients into interfaces and we replace them with mocks.